src/test/java/com/atlassian/db/replica/api/ThrottledCacheTest.java

```diff
@@ -113,11 +114,58 @@ public void shouldNotRobbedThreadReleaseLock() throws InterruptedException {
         assertThat(throwable).doesNotThrowAnyException();
     }
 
+    @Test
+    public void shouldntRobbedThreadUpdateValue() throws InterruptedException {
+        ThrottledCache<Long> cache = new ThrottledCache<>(clock, ofMillis(1500));
+
+        final WaitingWork firstThreadWork = asyncPutWithSlowSupplier(cache, 1);
+        clock.tick();
+        clock.tick();
+        cache.get(() -> 2L);
+        firstThreadWork.finish();
+        assertThat(cache.get()).isEqualTo(Optional.of(2L));
```

Comment:
good test, you want to enforce order of refreshed values.
You don't want that thread that comes first, but ended later, overridden the  value provided by second thread later, but finished earlier(without slow down).
👍